app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,27 @@
+class PostsController < ApplicationController
+  before_action :authenticate_user!, only: %i[new create]
+
+  def index
+    @posts = Post.all
+  end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(post_params)
```

Comment:
- I don't think this is right. A user is required but you are not using a user to create the post. Since we are using devise, we have access to the `current_user` object. Change to:
```ruby
@post = current_user.posts.build(post_params)
```